### PR TITLE
Implement AerDensityMatrix

### PR DESCRIPTION
**Pull request by derwind**
### Summary

This is a PR to implement AerDensityMatrix, which is a fine-tuned version of #1676.

### Details and comments

When this PR is approved, #1676 should be closed at the same time.

### CODE REVIEW

File: qiskit_aer/quantum_info/states/aer_densitymatrix.py [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+            else:
+                raise AerError(f'Input data is not supported: type={data.__class__}, data={data}')
+
+            self._aer_state = aer_state
+
+        super().__init__(data, dims=dims)
+
+        self._result = None
+        self._configs = configs
+
```
Comment (hhorii) [inline qiskit_aer/quantum_info/states/aer_densitymatrix.py:102]:
```suggestion

    def seed(self, value=None):
        """Set the seed for the quantum state RNG."""
        if value is None or isinstance(value, int):
            self._aer_state.set_seed(value)
        else:
            raise AerError(f'This seed is not supported: type={value.__class__}, value={value}')

```
seed method is missing.

File: qiskit_aer/quantum_info/states/aer_densitymatrix.py [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    def sample_memory(self, shots, qargs=None):
+        if qargs is None:
+            qubits = np.arange(self._aer_state.num_qubits)
+        else:
+            qubits = np.array(qargs)
+
+        aer_state = AerState(**self._aer_state.configuration())
+        aer_state.initialize(self._data, copy=False)
+        samples = aer_state.sample_memory(qubits, shots)
+        aer_state.close()
```
Comment (hhorii) [inline qiskit_aer/quantum_info/states/aer_densitymatrix.py:184]:
```suggestion
        self._aer_state.close()

        self._aer_state.renew()
        self._aer_state.initialize(self._data, copy=False)
        samples = self._aer_state.sample_memory(qubits, shots)
        self._data = self._aer_state.move_to_ndarray()
```

To keep a random generator in `AerState`, `sample_memory` needs the same `AerState` instance.

**Review by hhorii — COMMENTED** [review]
To get consistent results of `sample_memory()`, `seed()` needs to be overwritten and `self._aer_state` needs to be used for this method. 

https://github.com/Qiskit/qiskit-aer/commit/a71f17f6b89951542f5d55979c763c7cca23bf1d

File: qiskit_aer/quantum_info/states/aer_densitymatrix.py (hunk 1 above)
Comment (derwind) [reply qiskit_aer/quantum_info/states/aer_densitymatrix.py:102]: I had overlooked this. So Terra's QuantumState.seed is called and the effect of the call to seed() in test_aer_densitymatrix.py is inadequate. OK, I will fix this as you suggested.

File: qiskit_aer/quantum_info/states/aer_densitymatrix.py (hunk 2 above)
Comment (derwind) [reply qiskit_aer/quantum_info/states/aer_densitymatrix.py:184]: Thank you. I will correct it according to your suggestion and #1663.

**Review by hhorii — APPROVED** [review]
Now this looks good to me. Thank you for your contribution.

**Pull request merged**